visual_regression_tracker/visualRegressionTracker.py

```diff
@@ -97,21 +99,22 @@ def track(self, test: TestRun):
             if self.config.enableSoftAssert:
                 logging.getLogger(__name__).error(error_message)
             else:
-                raise Exception(error_message)
+                raise TestFailed(result.status, error_message)
 
 
 def _http_request(url: str, method: str, data: dict, headers: dict) -> dict:
     request = getattr(requests, method.lower())
     response = request(url, json=data, headers=headers)
     status = response.status_code
+    result = response.json()
 
     if status == 401:
-        raise Exception('Unauthorized')
+        raise ServerError({}, 'Unauthorized')
     if status == 403:
-        raise Exception('Api key not authenticated')
+        raise ServerError({}, 'Api key not authenticated')
     if status == 404:
-        raise Exception('Project not found')
+        raise ServerError({}, 'Project not found')
+    if status >= 400:
+        raise ServerError(result, result.get('message', 'An error occurred'))
```

Comment:
from server side we could get more fields 
this is the complete object 
```js
{
      path: request.url,
      stack: exception.stack,
      name: exception.name,
      message: exception.message,
      exception: exception,  // not formatted exception object
}
```

I was thinking about printing it all in console 
not sure how the first param is used here

Reply:
Thats a good point. I was coming from the direction that the stack trace overwhelmed the message and made it harder to consume, but there is definitely valuable info there too.
The change I made was to take the full details out of the print out and add it as a property on the exception object, but I'll update it to be in both.

Reply:
Done